velox/exec/LocalPartition.cpp

```diff
@@ -412,16 +406,10 @@ void LocalPartition::noMoreInput() {
 }
 
 bool LocalPartition::isFinished() {
-  if (numBlockedPartitions_ || !noMoreInput_) {
+  if (!futures_.empty() || !noMoreInput_) {
     return false;
   }
 
-  for (const auto& queue : queues_) {
```

Comment:
So LocalExchange holds a reference on the queue and it can continue to consume even after LocalPartition has finished? That's the purpose of this change? Thanks!

Reply:
That's right.